subprojects/docs/src/docs/userguide/authoring-builds/introduction/using_tasks.adoc

```diff
@@ -0,0 +1,393 @@
+
+Typically, a *lifecycle* tasks depends on many *actionable* tasks.
+
+[[sec:task_registration]]
+== Task registration and action
+
+Let's take a look at a simple "Hello World" task in a build script:
+
+====
+include::sample[dir="snippets/tutorial/hello/kotlin", files="build.gradle.kts[]"]
+include::sample[dir="snippets/tutorial/hello/groovy", files="build.gradle[]"]
+====
+
+In the example, the build script *registers* a single task called `hello`, and adds an *action* to it.
+
+If the tasks in the project are listed, the `hello` task is available to Gradle:
+
+[source]
+----
+$ ./gradlew app:tasks --all
+
+> Task :app:tasks
+
+------------------------------------------------------------
+Tasks runnable from project ':app'
+------------------------------------------------------------
+
+Other tasks
+-----------
+compileJava - Compiles main Java source.
+compileTestJava - Compiles test Java source.
+hello
+processResources - Processes main resources.
+processTestResources - Processes test resources.
+startScripts - Creates OS-specific scripts to run the project as a JVM application.
+----
+
+You can execute the task in the build script with `./gradlew hello`:
+
+====
+----
+$ ./gradlew hello
+include::{snippetsPath}/tutorial/hello/tests/hello.out[]
+----
+====
+
+When you run `./gradlew hello`, Gradle executes the `hello` task, which executes the *action* you've provided.
+The action is simply a block containing some code to execute: `println("Hello world!")`.
+
+== Task group and description
+
+The `hello` task from the previous section can be detailed with a *description* and assigned to a *group*.
+
+.build.gradle.kts
+[source,kotlin]
+----
+tasks.register("hello") {
+    group = "Custom"
+    description = "A lovely greeting task."
+    doLast {
+        println("Hello world!")
+    }
+}
+----
+
+Once the task is assigned to a group, it will be listed by `./gradlew tasks`:
+
+[source]
+----
+$ ./gradlew tasks
+
+> Task :tasks
+
+Custom tasks
+------------------
+hello - A lovely greeting task.
+----
+
+To view information about a task, use the `help --task <task-name>` command:
+
+[source]
+----
+$./gradlew help --task hello
+
+> Task :help
+Detailed task information for hello
+
+Path
+:app:hello
+
+Type
+Task (org.gradle.api.Task)
+
+Options
+--rerun     Causes the task to be re-run even if up-to-date.
+
+Description
+A lovely greeting task.
+
+Group
+Custom
+----
+
+As we can see, the `hello` task belongs to the `custom` group.
+
+[[sec:task_dependencies]]
+== Task dependencies
+
+You can declare tasks that depend on other tasks:
+
+====
+include::sample[dir="snippets/tutorial/intro/kotlin", files="build.gradle.kts[]"]
+include::sample[dir="snippets/tutorial/intro/groovy", files="build.gradle[]"]
+
+----
+$ gradle -q intro
+include::{snippetsPath}/tutorial/intro/tests/intro.out[]
+----
+====
+
+To add a dependency, the corresponding task does not need to exist:
```

Comment:
💅 "Not need to exist" is not really correct.  You say later: "The dependency of taskX to taskY may be declared before taskY is defined."  I think it would be better to move this sentence ahead of the example and delete the not need to exist one.